src/flexbuffer_cache.cpp

```diff
@@ -327,8 +327,9 @@ class flexbuffer_disk_cache
             // Does the source file's mtime match what we cached previously
             if( source_mtime != disk_entry->second.mtime ) {
                 std::string filepath_and_name = disk_entry->first;
-                // we use this as an exclusion condition. Configuration options can be changed all the time, we don't want to warn over those.
-                bool stale_game_data = *root_relative_source_path.begin() != std::filesystem::u8path( "config" );
+                // we use this as an exclusion condition. Configuration options can be changed all the time, we don't want to warn over those. Same for achievements.
+                bool stale_game_data = *root_relative_source_path.begin() != std::filesystem::u8path( "config" ) &&
+                                       *root_relative_source_path.begin() != std::filesystem::u8path( "achievements" );
```

Comment:
Would it perhaps be better to check the folders that we *do* care about (`data/` and `gfx/`) instead of playing whack-a-mole with all the folders we *don't*? The former ones are very easy to enumerate by simply examining the distribution archive (it only has `data/`, `doc/` `gfx/` `lang/`), whereas the latter is.. less straightforward, and harder to get right.

Reply:
I considered this approach but first consider that the `root_relative_source_path` when reading from data is not e.g. `data/json/some_file.json` but is instead `json/some_file.json`. Or e.g. `core/external_options.json`. 

There are multiple subfolders in there and I would consider it better that if new folders are automatically 'opted in' to the integrity checks, rather than relying on somebody remembering to update this piece of code. Thus if a new folder structure is devised at some point in the future it should start triggering false warnings immediately, and would be either be caught before the PR is merged or the author would hopefully still be around to make the connection.

False errors are, IMO, preferred to failing silently.